File: Python/items.py
import os

class Items:
    @staticmethod
    def init_items_file():
        if not os.path.exists('data/items.txt'):
            with open('data/items.txt', 'w') as f:
                f.write("Voiture\nAssiette\nAppartement\n")

    @staticmethod
    def list_items():
        try:
            with open('data/items.txt', 'r') as f:
                items = [line.strip() for line in f if line.strip()]
            if items:
                for item in items:
                    print(item)
            else:
                print("Aucun objet disponible.")
        except FileNotFoundError:
            print("Erreur : items.txt introuvable.")

    @staticmethod
    def item_exists(item):
        try:
            with open('data/items.txt', 'r') as f:
                return item in [line.strip() for line in f]
        except FileNotFoundError:
            return False

File: Ruby/items.rb
module Items
  def self.init_items_file
    return if File.exist?('data/items.txt')
    File.write('data/items.txt', "Voiture\nAssiette\nAppartement\n")
  end

  def self.list_items
    if File.exist?('data/items.txt') && !File.empty?('data/items.txt')
      puts File.read('data/items.txt').split("\n")
    else
      puts "Aucun objet disponible."
    end
  end

  def self.item_exists?(item)
    return false unless File.exist?('data/items.txt')
    File.read('data/items.txt').split("\n").include?(item)
  end
end